server/processors-alerts-publisher/src/main/java/org/hawkular/apm/processor/alerts/InvocationAlertsPublisherMDB.java

```diff
@@ -53,13 +55,41 @@
 
     @Override
     public void onMessage(Message message) {
-        logger.fragmentCompletionTimeReceived();
+        logger.invocationDetailsReceived();
         try {
             String data = ((TextMessage) message).getText();
-            List<CompletionTime> items = mapper.readValue(data, new TypeReference<List<CompletionTime>>() {});
-            publisher.publish(items, "FragmentCompletionTime");
+            List<NodeDetails> items = mapper.readValue(data, new TypeReference<List<NodeDetails>>() {});
+            List<Event> events = items.stream().filter(nd -> nd.isInitial())
+                    .map(InvocationAlertsPublisherMDB::toEvent)
+                    .collect(Collectors.toList());
+            if (!events.isEmpty()) {
+                publisher.publish(events);
```

Comment:
IIRC, the `#publish(List)` iterates over the list. It was thought as mostly a helper method, but on this case, it might be best to call `#publish(Event)` individually and spare the list.